src/common/MetaPreview/MetaPreview.js

```diff
@@ -194,6 +209,15 @@ const MetaPreview = ({ className, compact, name, logo, background, runtime, rele
                         :
                         null
                 }
+                {
+                    <ActionButton
```

Comment:
you need to check for `typeof showHref === 'string' && compact` to make sure this button is rendered only if there is a valid href and user is in discover